### DISCUSSION TITLE
PnP Search Results Web Part Not Showing List Values

### DISCUSSION THREAD

**Discussion by RamBalani** [microsoft-search/pnp-modern-search#1981] [Q&A] [answered]
The managed property of  our custom (from blank) list columns (after 'Title') were all created via 'Add from Existing Site Column'  meaning Sharepoint automatically generated, and these auto Managed Property are ALL  showing up in Site schema.
PnP Search web part confirmed to  work (finds matching PnP search results based on keywords entered) but PnP Search Results   only shows Title column. 
I suspect a mapping crawled to managed issue but all Managed properties from list site columns generated after 2-3 list items were added are greyed in the Site Schema where Managed and Crawled properties are shown.
The PnP Search results page are shown on one of these screen shots. PLEASE HELP!

=========================================
[PnP Search Result Showing Realization keyword -not the List Value but Managed Property Name.pdf](https://github.com/microsoft-search/pnp-modern-search/files/8379156/PnP.Search.Result.Showing.Realization.keyword.-not.the.List.Value.but.Managed.Property.Name.pdf)
----------------------
![PnP Seatch Results Web Part Shows Managed Property](https://user-images.githubusercontent.com/37209670/160799289-723aef2b-bdbd-4900-9d2a-246c5482dd67.jpg)
-----------------------
![Site Schema-Crawled Properties Autocreated for List Columns](https://user-images.githubusercontent.com/37209670/160799342-5238097d-9c5e-4fdd-a85a-599453964013.jpg)
-----------------------
![Site Schema-Managed Properties Autocreated for List Columns](https://user-images.githubusercontent.com/37209670/160799383-ab8b566d-1e98-4a3c-a5b4-5a6a74721197.jpg)

**Comment by wobba** [comment]
The "API" to list properties is not a real API so what we get is a best effort. If you don't see it, manually type it into the "Selected properties" box and hit ENTER. Then it will appear in the dropdown list.

**Reply by wobba** [reply]
@RamBalani type in the name of the managed property you want returned.

**Comment by RamBalani** [comment]
Thanks, what exactly to type?On Mar 30, 2022 1:22 PM, Mikael Svenson ***@***.***> wrote:
The "API" to list properties is not a real API. If you don't see it, manually type it into the "Selected properties" box and hit ENTER. Then it will appear in the dropdown list.

**Comment by RamBalani** [comment]
Am confused—does NOT the 1st   the screen shot attached here ALREADY now show the Selected Properties include my desired properties?

 

These Selected Properties—as I understand it – are what I need to incorporate into my Managed Columns-and I do !

But they DO NOT APPEAR on the PnP Search result page- THIS IS THE CRUX OF MY ISSUE!

**Reply by wobba** [reply]
Please don't reply via e-mail to discussions as the threading breaks badly (unfortunately). Re-read your issue now, and my mistake on the mp not showing. So I don't really understand your issue, meaning what is not working? And the schema will look greyed out, that's just the UX. To add columns to the details layout that is explained at the documentation https://microsoft-search.github.io/pnp-modern-search/usage/search-results/layouts/#details-list

If that's not the issue you need to add a more detailed explanation about what you cannot get to work.

**Reply by RamBalani** [reply]
Apologies-my bad-we're new to GitHub. I have reviewed a few days back the link you suggested here.
The issue is as follows;
1-We're referencing  regular columns on our list, columns were created via 'Add from Existing Site Column'
2-We understand crawled properties are AUTOMATICALL MAPPED to Managed Properties- we have below (1/5) as shown 'ISO13485ClauseCategoryOWSTEXT' 
3-Issue is the List item data not showing up when we include in the Managed column , i.e entered as column 2 to display on the PnP Search results page
4-Only the Title column is shown on the results page
5-We confirm with Search Box keyword, e.g. 'quality' and indeed it finds only one list item that meet that search keyword, i.e. seems PnP Search Box has been connected appropriately to the PnP Search results page --
6-BUT what shows up on results page is the same Managed Property name , i.e. "ISO13485ClauseCategoryOWSTEXT"  AND NOT THE ACTUAL LIST ITEM DATA 

Attachments:

1-Shows the source List
![Test List Items beign Searched inside PnP Search Results Page](https://user-images.githubusercontent.com/37209670/160996894-27067a39-cd11-4d8f-9a5c-8057384a2546.jpg)

2-Selected properties screen -we test with the first one

**Reply by RamBalani** [reply]
Sorry-more Attachments-pressed the wrong key by mistake.
2-Selected Properties
![Selected Properties Already done](https://user-images.githubusercontent.com/37209670/160997241-643ed82d-d6ed-452e-9137-88c5f0bbd378.jpg)

3-Managed Columns
![Selected Properties Already -One Chosen in Managed COLUMN as test](https://user-images.githubusercontent.com/37209670/160997437-acc9db78-1a06-4792-a9cd-15c36120311d.jpg)

4-Results page shows the Managed Property name NOT THE LIST VALUE based on 'quality' search box entered
![PnP Search Results  Shows Finding Correct List item-contains the word quality](https://user-images.githubusercontent.com/37209670/160997585-89c9e3f6-e39e-4fcb-b209-27fcf72eb6c1.jpg)

FOLLOW what the problem is ? i.e. Am not able to display the actual list value for the List item that meet search box criteria but instead PnP Search Page shows the **NAME of the MANAGED PROPERTY**
WHAT AM I DOING WRONG ? SHOULD I BE INCLUDING A HANDLE BAR?

**Reply by wobba** [reply]
You need to use handlebar notation. `{{ISO13485ClauseCategoryOWSTEXT}}`, same as you would in any template. If you have mapped it to a slot take a look at how the Title handlebar template looks like.

**Comment by RamBalani** [comment]
That was not the issue it turns out, i.e. not handlebars absence.
 I merely  re-indexed the list data source and like magic now shows up.
Besides- I believe Handle Bars are not mandatory if I recall correctly (could be wrong?) -read that somewhere few days back.
That said--WITHOUT USING HANDLEBARS-see Screen shots--the List Data Source columns  PnP Results page NOW shows the List source upon loading  up as expected.
HOWEVER---(!!!) New ISSUE has  popped up - Search Box-upon entry of the same keyword term 'quality' FAILS to filter the Results Page which it successfully did before (OUCH) , i.e. seems the connection between Pnp Search Box with PnP Results page has been broken. How to fix? Suggest documentation should include some obvious or apparent mistakes maybe?  Not sure am making one-see screen shots.
And Yes-- I deleted the old Search Box-replaced with a Search box replacement, re-configured as before I believe--but still fails to find only List row with 'quality' term. What are possible causes for Search Results page not getting triggered  by search term entered?
**ATTACHMENTS:**
1-Shows List Data Source Columns-no Handlebar was used
![Shows PnP Search Results Available Connection Setting-List Col Showing without handlebar](https://user-images.githubusercontent.com/37209670/161018260-240dbd7b-caed-44b7-a3ab-d6f740f934db.jpg)

2-Shows 'quality' Search fails 
![PnP Search Box-Quality Search Entered-Fails to Filter Results Page](https://user-images.githubusercontent.com/37209670/161018422-676c6bd5-6043-4ee1-b15b-2e17031b456d.jpg)

3-Current PnP Search BOX Configuration Setting
![Search Box Configuration Settings](https://user-images.githubusercontent.com/37209670/161018509-af8f800c-e6c7-4477-8e1f-09f8dcba241e.jpg)

Many thanks for the attention-hope am not being a 'pain'-- but I really want this to work! It's awesome, once working. 🚀🍳

**Comment by RamBalani** [comment, marked as answer]
Issue fixed-- I re-configured PnP Search Results Available Connections, i.e my bad! 
PnP Search Box somehow was missing

![Issue Fixed-Search Box Filters Search Results-Reconfigured PnP Search Results Setting](https://user-images.githubusercontent.com/37209670/161023832-8a701826-0630-4fc9-bbb6-480c2f4a3559.jpg)

Thanks for the group's patience-thanks Mikael.
**One parting question:**
Can the PnP Results Page be configured such that after a SearchBox search--user selects a specific list item row and after selecting-triggers a dynamic linking with another list on the same Modern Page? That's our send goal:
Search List #1 --> Select 1 or more Rows from List #1  --> then  Dynamically filter List #2  on the SAME MODERN PAGE?

**Comment by RamBalani** [comment]
Ok will do, from thevsound of this a list in searh pnp web part will notdo dynamic list?On Mar 30, 2022 3:05 PM, Mikael Svenson ***@***.***> wrote:
@RamBalani type in the name of the managed property you want returned.

**Comment by RamBalani** [comment]
This has not resolved the issue-am not sure you all follow what is going on.

See screen shots I have provided.

 

 

 

 

Best Regards,

Ram Balani

CEO – eSTARHelper LLC /FDASmart Inc.

 <http://www.estarhelper.com> www.estarhelper.com
